deps/rabbitmq_cli/lib/rabbitmq/cli/diagnostics/commands/check_for_quorum_queues_without_an_elected_leader_command.ex

```diff
@@ -0,0 +1,105 @@
+## This Source Code Form is subject to the terms of the Mozilla Public
+## License, v. 2.0. If a copy of the MPL was not distributed with this
+## file, You can obtain one at https://mozilla.org/MPL/2.0/.
+##
+## Copyright (c) 2007-2025 VMware, Inc. or its affiliates.  All rights reserved.
+
+defmodule RabbitMQ.CLI.Diagnostics.Commands.CheckForQuorumQueuesWithoutAnElectedLeaderCommand do
+  alias RabbitMQ.CLI.Core.DocGuide
+
+  @behaviour RabbitMQ.CLI.CommandBehaviour
+
+  import RabbitMQ.CLI.Core.Platform, only: [line_separator: 0]
+
+  def switches(), do: [across_all_vhosts: :boolean]
+
+  def scopes(), do: [:diagnostics]
+
+  def merge_defaults(args, opts) do
+    {args, Map.merge(%{across_all_vhosts: false, vhost: "/"}, opts)}
+  end
+
+  use RabbitMQ.CLI.Core.AcceptsOnePositionalArgument
+  use RabbitMQ.CLI.Core.RequiresRabbitAppRunning
+
+  def run([pattern] = _args, %{node: node_name, vhost: vhost, across_all_vhosts: across_all_vhosts_opt}) do
+    vhost = if across_all_vhosts_opt, do: :across_all_vhosts, else: vhost
+
+    case :rabbit_misc.rpc_call(node_name, :rabbit_quorum_queue, :leader_health_check, [pattern, vhost]) do
+      [] ->
+        :ok
+
+      unhealthy_queues_or_error ->
+        {:error, unhealthy_queues_or_error}
+    end
+  end
+
+  def output(:ok, %{node: node_name, formatter: "json"}) do
+    {:ok,
+     %{
+       "result" => "ok",
+       "message" =>
+         "Node #{node_name} reported all quorum queue leaders as healthy"
+     }}
+  end
+
+  def output(:ok, %{silent: true}) do
+    {:ok, :check_passed}
+  end
+
+  def output(:ok, %{node: node_name}) do
+    {:ok, "Node #{node_name} reported all quorum queue leaders as healthy"}
+  end
+
+  def output({:error, unhealthy_queues}, %{node: node_name, formatter: "json"}) when is_list(unhealthy_queues) do
+    {:ok, :check_passed,
+     %{
+       "result" => "error",
+       "queues" => unhealthy_queues,
+       "message" => "Node #{node_name} reported unhealthy quorum queue leaders"
+     }}
+  end
+
+  def output({:error, unhealthy_queues}, %{silent: true}) when is_list(unhealthy_queues) do
+    {:ok, :check_passed}
+  end
+
+  def output({:error, unhealthy_queues}, %{vhost: _vhost}) when is_list(unhealthy_queues) do
+    lines = queue_lines(unhealthy_queues)
+
+    {:ok, :check_passed, Enum.join(lines, line_separator())}
```

Comment:
@Ayanda-D if this is an error, why does this return an `:ok` and `:check_passed`?

As things stand right now, this command will always exit with a code of zero. That's not at all the convention we have for all the other health checks.

It should have been

```elixir
{:error, :check_failed}
```

or one of its variations.